agent/jobinfo.go

```diff
@@ -27,6 +28,29 @@ func GetJobInfo() (*JobInfo, error) {
 	return envToJobInfo(env)
 }
 
+func GetJobInfoFromFile(file string) (*JobInfo, error) {
+	var data []byte
+	d, err := os.ReadFile(file)
+	if err != nil && !os.IsNotExist(err) {
+		return nil, err
+	}
+	// Accepts ErrNotExist.
+	// When ErrNotExist is returned, it means that `job_started` was not called in a workflow.
+	// Even in this case, slack notification will run.
+	data = d
```

Comment:
I think it is implementation-dependent to expect `d` to be nil when `os.ReadFile` returns an error.
But, in this case, it's not a problem.